sli/parser.h

```diff
@@ -61,6 +61,7 @@ class Parser
 public:
   Parser();
   Parser( std::istream& );
+  ~Parser() { if (s) delete s; }
```

Comment:
According to *The C++ Programming Language*, 4th ed, §11.2, "[a]pplying `delete` to the `nullptr` has no effect." So the `if` is not needed here:
```suggestion
  ~Parser() { delete s; }
```

In general, we require braces around single-line blocks (as a consequence of the OpenSSH goto-goto bug some years back).